### ISSUE TITLE
[Support]: Error 404 when attempting to view snapshots and clips from push notification

### ISSUE THREAD

**Issue by C0nvictions** [blakeblackshear/frigate#4689]
### Describe the problem you are having

To preface, my experience with unraid, home assistant, and VMs is just the last couple weeks that I've been working to set up a home server for a security camera system. I'm running a VM with Home Assistant OS inside of Unraid. I have frigate installed as a docker within Unraid to capture video. In the Home Assistant VM, I'm using frigate proxy.

Notifications have been set up using the frigate blueprint. I currently get notifications, but trying to tap "view snapshot" or "view clip" results in a 404 error. I can still get to the snapshots and clips from within frigate proxy, however. From what I've seen online, this issue is often a result of the frigate integration not being installed in HACS, but I do have it installed.

I appreciate any help anyone can provide!

### Version

Frigate Proxy v. 1.3

### Frigate config file

```yaml
mqtt:
  host: xxxxxxxxxxxx
  user: xxxxxxxx
  password: xxxxxxxxxxx
  
objects:
  track:
    - person

rtmp:
  enabled: true

record:
  enabled: True
  events:
    retain:
        default: 7

snapshots:
      enabled: True
      bounding_box: true
      retain:
        default: 7

birdseye:
  enabled: True
  mode: continuous
  
cameras:
  backdoor: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx # <----- Update for your camera
          roles:
            - detect
            - rtmp
        - path: rtsp://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx # <----- Update for your camera
          roles:
            - record
         
    detect:
      width: 2560 # <---- update for your camera's resolution
      height: 1440 # <---- update for your camera's resolution
    
    motion:
      mask:
      - 2160,0,2213,1440,2560,1440,2560,0
      - 2560,0,1547,368,2086,168,2105,1413
            
  bedroom: # <------ Name the camera
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxxxxxxxxxxxxxxxxxxxxxxx # <----- Update for your camera
          roles:
            - detect
            - rtmp
        - path: rtsp://xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx # <----- Update for your camera
          roles:
            - record
        
    detect:
      width: 2560 # <---- update for your camera's resolution
      height: 1440 # <---- update for your camera's resolution
```

### Relevant log output

```shell
Not sure what this would be.
```

### FFprobe output from your camera

```shell
Not sure what this would be or if it'd be relevant.
```

### Frigate stats

```json
Not sure what this is.
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Tapo C320WS

**Comment by NickM-27** [comment]
What did you set as your BASE URL in the blueprint ?

**Comment by C0nvictions** [comment]
I've tried using the ip for home assistant and also tried leaving it blank. Both times I get 404 errors.

**Comment by NickM-27** [comment]
What happens if you try the url manually?

**Comment by C0nvictions** [comment]
Which URL are you suggesting to try?

**Comment by NickM-27** [comment]
http://ha_ip:8123/api/frigate/notifications/{{id}}/snapshot.jpg

**Comment by C0nvictions** [comment]
I'm pretty sure that address opened in chrome window on its own when I input the HA address in the BASE URL field. The chrome window just displayed a 404 error.

**Comment by NickM-27** [comment]
If a known good event id is used and it 404s then something is likely wrong with the integration

**Comment by NickM-27** [comment]
try http://ha_ip:5000/API/events/{event_id}/snapshot.jpg and see if that resolves

(Port 5000 needs to be enabled in the frigate addon settings and use the same event id from before)

**Comment by C0nvictions** [comment]
I don't have the frigate addon -- I'm using frigate proxy. Trying your other URL suggestion gives a 403 error, though.

**Comment by NickM-27** [comment]
Then it'll be http://frigate_ip:5000/API/events/{event_id}/snapshot.jpg

**Comment by C0nvictions** [comment]
That brings me to a page that says "event not found"

**Comment by NickM-27** [comment]
That explains why it wasn't working in the integration then. Will need to try using an event id that is valid. You can get a url from the download snapshot button by right clicking it b

**Comment by C0nvictions** [comment]
I got a url from there and it's:

http://ha_ip:8123/api/hassio_ingress/event_id/snapshot.jpg

**Comment by NickM-27** [comment]
I'd recommend from frigate ui without ingress then work up from there.

**Comment by C0nvictions** [comment]
The url from frigate (running as a docker in unraid) is: http://frigate_url:5000//api/events/event_id/snapshot.jpg

I'm not sure what to do with that information, though.

**Comment by NickM-27** [comment]
If it works and shows the snapshot in the browser, then cut out the event id and paste it in http://ha_ip:8123/api/hassio_ingress/event_id/snapshot.jpg and see if that works

**Comment by C0nvictions** [comment]
That one actually doesn't work. I think that was the original link I said I got, but it looks like I left some out. It's actually:

http://ha_ip:8123/api/hassio_ingress/long_string_that_seems_to_identify_my_frigate_poxy_instance//api/events/event_id/snapshot.jpg

**Comment by NickM-27** [comment]
And that one works too right?

**Comment by C0nvictions** [comment]
yes, that one works.

**Comment by NickM-27** [comment]
Great, then use the same event ID in http://ha_ip:8123/api/frigate/notifications/{{id}}/snapshot.jpg does that work?

**Comment by C0nvictions** [comment]
Nope, error 404.

**Comment by NickM-27** [comment]
Then there you go, the integration isn't setup correctly. What did you use as the frigate address when setting it up?

**Comment by C0nvictions** [comment]
I don't recall there being any setup options for the integration -- is there a file I need to modify?

**Comment by NickM-27** [comment]
When you setup the integration it asks for the address of frigate so it can connect to it. You'll need to remove and re-add the integration.

also, to be clear since some people don't know, the integration and the addon are two completely different things

**Comment by C0nvictions** [comment]
Yeah, I'm aware. The frigate proxy addon asked for an address. The HACS integration never asked me for anything. I just removed and re-downloaded it. It just prompted me to restart home assistant, but no prompt for an address.

**Comment by NickM-27** [comment]
Right but did you set it up inside the devices & actions section of HA settings?

**Comment by NickM-27** [comment]
> ... well, THERE'S my problem. I didn't realize there was a direct HA integration. I always thought the "integration" was just the HACS one.
> 
> 
> 
> I have to leave now, but hopefully this fixes things. I'll update later.

I think you're still confused. The HACS integration is the only integration but you of course still need to actually set it up for it to do anything.

**Comment by C0nvictions** [comment]
![Screenshot_20221214_083605_Home Assistant](https://user-images.githubusercontent.com/81660340/207655254-91a3234a-ed0f-4a60-b798-17b8f0e84767.jpg)

![Screenshot_20221214_083536_Home Assistant](https://user-images.githubusercontent.com/81660340/207655322-28b1b0d4-f43c-4d6a-ac94-34fe8ec8bc13.jpg)


The first screenshot is what I'm calling the HA integration, the second is the HACS integration.

**Comment by NickM-27** [comment]
They are the exact same thing, you couldn't get the first one without having the second one

**Comment by NickM-27** [comment]
just so it's clear for others that may have the same confusion

per the installation instructions: https://github.com/blakeblackshear/frigate-hass-integration#installation

![Screenshot 2022-12-14 094607](https://user-images.githubusercontent.com/14866235/207656358-91f672f8-cb4f-4a8d-b146-df6e38089bb9.png)

**Comment by C0nvictions** [comment]
Thanks for clearing that up. I think I may have done it the right way at some point, but my MQTT settings were messed up and I ended up uninstalling everything and retracing my steps poorly.

At any rate, significant progress, but everything still doesn't seem to be quite right. When I get notifications, they have a snapshot preview built in. However, the notification "view clip" button and tapping the notification itself bring me to the video only if i set an internal home assistant URL. I'm using tailscale to access home assistant, so I set internal the same as external.

If I don't set an internal URL, I get the popup that says "unable to connect to Home Assistant" with "settings," "refresh external URL" and "wait" options.

Even with internal URL set (again, same as my external URL), the "snapshot" button on the notification still goes to the "unable to connect" popup. What's more, even though the "clip" button starts playing the clip, about 3/4 of the way through, the "unable to connect" popup shows up. The video continues to play, though, if I tap "wait." The popup then comes back again after several seconds.
